clippy_lints/src/panic_unimplemented.rs

```diff
@@ -96,8 +96,7 @@ impl<'tcx> LateLintPass<'tcx> for PanicUnimplemented {
         if_chain! {
             if let ExprKind::Block(ref block, _) = expr.kind;
             if let Some(ref ex) = block.expr;
-            if let Some(params) = match_function_call(cx, ex, &paths::BEGIN_PANIC)
-                .or_else(|| match_function_call(cx, ex, &paths::BEGIN_PANIC_FMT));
+            if let Some(params) = match_panic_call(cx, ex);
             then {
                 let span = get_outer_span(expr);
                 if is_expn_of(expr.span, "unimplemented").is_some() {
```

Comment:
This check doesn't seem to work for the core macros. But I don't see why.

Reply:
Time for some println debugging!

Reply:
It also doesn't detect `std::todo` either. The problem apparently isn't `core::`, but just a qualified name.

Reply:
Since that seems a separate issue, it's probably good to just change the test to avoid qualifying the macro:

```rust
extern crate core;
 
fn core_versions() {
    use core::{panic, todo, unimplemented, unreachable};
    panic!();
    todo!();
    unimplemented!();
    unreachable!();
}
```

Reply:
Huh, weird. Thanks for figuring it out :)